### PR TITLE
Remove Outdated Code (Part 3)

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

Related PRs: #1397, #1411, #1414

*Description of changes:*

- Removed all outdated HPO code except for GridSearcher (Will migrate to LocalGridSearcher in follow-up PR)
- Removed FIFOScheduler
- Removed RandomSearcher
- Removed Bayesopt Searcher
- Refactored autogluon.vision to use LocalSequentialScheduler and LocalSearcher
- Follow-up PR will remove ConfigSpace, only very few usages remain in package.
- Minor fix to Pseudo-labelling when all test data was pseudo-labelled (previously crashed).
- Majorly sped up tabular unit tests (particularly pseudo-labelling)
- Removed unnecessary gpu instance requirement for feature and core unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1431-21 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1431/21/index.html

**Review by yinweisu — APPROVED** [review]

File: core/src/autogluon/core/scheduler/seq_scheduler.py [hunk 1]
```diff
@@ -86,15 +87,15 @@ class LocalSequentialScheduler(object):
...
         self.training_history = None
         self.config_history = None
-        self._reward_attr = kwargs['reward_attr']
+        self._reward_attr = reward_attr
         self.time_attr = kwargs.get('time_attr', None)
-        self.resource = kwargs['resource']
+        self.resource = resource
         self.max_reward = kwargs.get('max_reward', None)
-        self.searcher: BaseSearcher = self.get_searcher_(searcher, train_fn, search_space=search_space, **kwargs)
+        self.searcher: LocalSearcher = self.get_searcher_(searcher, train_fn, search_space=search_space, **kwargs)
```
Comment (gradientsky) [inline core/src/autogluon/core/scheduler/seq_scheduler.py:98]: Why replace `BaseSearcher` with less abstract version of this class? I'm thinking about a case when DistributedSearcher is added and LocalSeracher is too specific here.

**Comment by gradientsky** [comment]
I don't see any updates to `core/src/autogluon/core/searcher/searcher_factory.py` - not sure if it was done before, but there were a few `bayesopt` and `random` references in `_scheduler_presets` there.

File: docs/index.rst [hunk 2]
```diff
@@ -88,12 +77,6 @@ Advanced Topics
    tutorials/text_prediction/index
    tutorials/tabular_prediction/tabular-multimodal
    tutorials/cloud_fit_deploy/index
-   tutorials/course/index
    api/autogluon.predictor
-   api/autogluon.core
-   api/autogluon.core.space
```
Comment (gradientsky) [inline docs/index.rst:94]: Spaces documentation is still useful: https://auto.gluon.ai/stable/api/autogluon.core.space.html#module-autogluon.core.space

File: docs/api/index.rst [hunk 3]
```diff
@@ -12,22 +12,8 @@ describing what they are and what they do. The APIs are grouped into the followi
 
       Automatically train/deploy predictors with 1 line of code for text, image, or tabular data.
 
-   .. card::
-      :title: Search Space
-      :link: autogluon.core.space.html
-
-      Hyperparameter search spaces of candidate hyperparameter values to consider.
```
Comment (gradientsky) [inline docs/api/index.rst:19]: I don't think we should remove Spaces documentation: the existing documentation is useful: https://auto.gluon.ai/stable/api/autogluon.core.space.html#module-autogluon.core.space

File: core/src/autogluon/core/scheduler/seq_scheduler.py (hunk 1 above)
Comment (Innixma) [reply core/src/autogluon/core/scheduler/seq_scheduler.py:98]: Unclear what BaseSearcher would look like atm. Can revisit after finish removing outdated code.

Comment (Innixma) [reply core/src/autogluon/core/scheduler/seq_scheduler.py:98]: LocalSearcher does not depend on BaseSearcher, I'm deleting BaseSearcher in this PR.

File: docs/index.rst (hunk 2 above)
Comment (Innixma) [reply docs/index.rst:94]: Removing as it referenced configspace in docs. We can add back afterwards in follow-up.

File: docs/api/index.rst (hunk 3 above)
Comment (Innixma) [reply docs/api/index.rst:19]: Removing as it referenced configspace in docs. We can add back afterwards in follow-up.

File: Jenkinsfile [hunk 4]
```diff
@@ -26,6 +26,9 @@ setup_pip_venv = """
 setup_mxnet_gpu = """
     python3 -m pip install mxnet-cu101==1.7.0
     export MXNET_CUDNN_AUTOTUNE_DEFAULT=0
+    nvidia-smi
+    ls /usr/local/
```
[the remark this answers is not in the document]
Comment (Innixma) [reply Jenkinsfile:30]: will add in follow-up PR

**Review by gradientsky — APPROVED** [review]
